### PR TITLE
Add support for meta in Spider Contracts

### LINKED ISSUE #6468: Add support for meta in Spider Contracts
## Summary

Today we support `cb_kwargs` but we have scenarios where the data is passed using `meta`.

## Motivation

I have some spiders that use `meta` to store information, e.g: `cookiejar`

## Describe alternatives you've considered

Use callback args.

### PR DESCRIPTION
**Pull request by Laerte**
close #6468

### CODE REVIEW

File: docs/topics/contracts.rst [hunk 1]
```diff
@@ -46,6 +46,14 @@ This callback is tested using three built-in contracts:
 
     @cb_kwargs {"arg1": "value1", "arg2": "value2", ...}
 
+.. class:: MetadataContract
```
Comment (Laerte) [inline docs/topics/contracts.rst:49]: Above in this file we have:

https://github.com/scrapy/scrapy/blob/6ce0342beb1a5b588f353e52fe03d5e0ec84d938/docs/topics/contracts.rst?plain=1#L29

I'm not sure if this statement is correct since we just just are listing the default contracts and `cb_kwargs` is not used in the snippet:

https://github.com/scrapy/scrapy/blob/6ce0342beb1a5b588f353e52fe03d5e0ec84d938/docs/topics/contracts.rst?plain=1#L23-L26

I followed the pattern and added the `MetadataContract` to the list.

File: docs/topics/contracts.rst [hunk 2]
```diff
@@ -46,6 +46,14 @@ This callback is tested using three built-in contracts:
...
+
+    This contract (``@meta``) sets the :attr:` meta <scrapy.Request.meta>`
```
Comment (Laerte) [inline docs/topics/contracts.rst:51]:
```suggestion
    This contract (``@meta``) sets the :attr:`meta <scrapy.Request.meta>`
```

File: docs/topics/contracts.rst (hunk 1 above)
Comment (Gallaecio) [reply docs/topics/contracts.rst:49]: I would reword that to:

> You can use the following contracts:

Also, I think we should replace amazon.com with example.com.

**Review by Gallaecio — APPROVED** [review]

File: docs/topics/contracts.rst (hunk 1 above)
Comment (Laerte) [reply docs/topics/contracts.rst:49]: Done, thanks.

File: scrapy/contracts/default.py [hunk 3]
```diff
@@ -35,6 +35,20 @@ def adjust_request_args(self, args: Dict[str, Any]) -> Dict[str, Any]:
         return args
 
 
+class MetadataContract(Contract):
+    """Contract to key metadata arguments for the request.
```
Comment (Laerte) [inline scrapy/contracts/default.py:39]:
```suggestion
    """Contract to set metadata arguments for the request.
```

**Review by wRAR — APPROVED** [review]
